vulcanai/models/metrics.py

```diff
@@ -1012,3 +1014,85 @@ def cross_validate(network, data_loader, k, epochs,
             logger.info(all_results)
             return all_results
 
+    def conduct_sensitivity_analysis(self, network, data_loader, filename,
+                                     features, cutoff=20):
+        """
+        Will conduct tests to figure out directionality of features by finding all the unique feature values present in
+        the dataset and setting the feature for all examples to every unique value of that feature.
+
+        Parameters
+        ----------
+            network : BaseNetwork
+                Network descendant of BaseNetwork.
+            data_loader : DataLoader
+                A DataLoader object which contains all the data
+            filename: string
+                The name of the file to save the test results to
+            cutoff: int
+                Maximum number of unique feature values for a particular feature that will be tested
+
+        Returns
+        -------
+        #TODO
+        """
+
+        if network._num_classes is None or network._num_classes == 0:
+            raise ValueError('There\'s no classification layer')
+
+        dataset = data_loader.dataset
+
+        test_df = None
+        col_headers = None
+
+        for feature in range(len(features)):
+            feature_val_list = dataset[:][0][:, feature].tolist()
+
+            unique_feature_values = list(np.unique(feature_val_list))
+
+            if len(unique_feature_values) > cutoff:
+                print("================================")
+                print("Cutting off number of features to be tested for feature "
+                      "{} from {} to {}".format(features[feature],
+                                                len(unique_feature_values),
+                                                cutoff))
+                print("================================")
+                unique_feature_values = unique_feature_values[0::int(
+                        len(unique_feature_values) / cutoff)]
+            dataset_temp = copy.deepcopy(dataset)
```

Comment:
instead of doing a deep copy of `dataset`. Why not deep copy the data loader dataset? then you store one less instance of dataset in memory because you already treat it as immutable

Reply:
I don't get what you mean? Instead of this line I do data_loader_temp = copy.deepcopy(data_loader)? How will that save memory?

Reply:
you set `dataset = dataloader.dataset` and then deep copy `dataset`. Why not just deep copy from `deepcopy(data loader.dataset)` instead? then you don't need to store the data in the local `dataset` variable

Reply:
I use dataset in other areas in the code (~ lines 1071, 1051). I could just update these to using dataloader.dataset instead of creating the copy beforehand if that's better?

Reply:
yes I think so - that's what other parts of metrics uses when you look around. @nirtiac just want to get a third opinion from you that this is ok?